### ISSUE TITLE
Wrong layerset-slicing on the edges of IfcRelConnectsPathElements

### ISSUE THREAD

**Issue by johanrd** [IfcOpenShell/IfcOpenShell#654]
1. Convert [654--wrong-layerset-slicing.ifc](https://github.com/IfcOpenShell/files/blob/master/654--wrong-layerset-slicing.ifc) with `--enable-layerset-slicing` and experience that the the material "yellow paint" is being hijacked by the material "Lath" in the connection between the two walls:
<img width="250" alt="Screenshot 2019-08-12 at 14 25 53" src="https://user-images.githubusercontent.com/4601554/62865304-66005380-bd0e-11e9-83d5-c780e397ceb0.png">

This bug seems to be caused by line `#56362 = IfcRelConnectsPathElements`, since the file renders as expected when this line is commented out:
```
#56362= IFCRELCONNECTSPATHELEMENTS('2LhCnUaHzBnuPXd3425Ve4',#32,$,$,$,#56354,#54393,(),(),.ATEND.,.ATSTART.);
```

<img width="250" alt="Screenshot 2019-08-12 at 14 26 23" src="https://user-images.githubusercontent.com/4601554/62865590-330a8f80-bd0f-11e9-9f19-ee4a53568946.png">

**Comment by aothms** [comment]
Thanks, this is fixed in 16ce1f0823ae596e773970a436c553c32db0c3e9

Folds are made at the connections. But in this case it's not necessary because of the way the walls are joined. Before only the wall axis lengths were used to determine if "folds" had to be made. Now that's determined based on the offset direction and other wall axis direction.

**Comment by johanrd** [comment]
Nice! I can verify that 16ce1f0 fixes the case for [654--wrong-layerset-slicing.ifc](https://github.com/IfcOpenShell/files/blob/master/654--wrong-layerset-slicing.ifc).

However, not closing the issue, as I found a similar case from the same source file:

1. Convert [654--wrong-layerset-slicing--2.ifc](https://github.com/IfcOpenShell/files/blob/master/654--wrong-layerset-slicing--2.ifc) with `--enable-layerset-slicing` and experience that the the material "White plaster" is being hijacked by the material "Lath" in the connection between the two walls:
<img width="250" alt="Screenshot 2019-08-16 at 17 14 38" src="https://user-images.githubusercontent.com/4601554/63178308-1392a000-c04a-11e9-9d0c-ca5a4d4adee3.png">

Removing line `#202153 = IfcRelConnectsPathElements` from the file (the wall that is connected on the other side), makes the wall render as expected:
```
#202153= IFCRELCONNECTSPATHELEMENTS('1R2yAVXGSrTtw1rvuKi_Ii',#32,$,$,$,#202140,#198156,(),(),.ATSTART.,.ATEND.);
```

<img width="250" alt="Screenshot 2019-08-16 at 17 14 48" src="https://user-images.githubusercontent.com/4601554/63178420-49378900-c04a-11e9-99f1-6bb129d7ac2e.png">

**Comment by aothms** [comment]
Thanks, this was indeed related. The filter on parallel connection caused the wrong other end point to be selected as the folding point. This is now fixed in 75b24c23521311258b8045bb457ec2ab333a5ef4

**Comment by johanrd** [comment]
Nice, thanks, I can verify that 75b24c2 fixes this. Closing.

We don't use `--enable-layerset-slicing` just yet, but we will look into it. Anyways, we are dependent on a better fallback handling for when layerset-slicing isn't enabled (ref. #649).
